cmd/secrets-provider/main.go

```diff
@@ -51,51 +51,42 @@ func main() {
 		printErrorAndExit(messages.CSPFK009E)
 	}
 
-	// Configure exponential backoff
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 2 * time.Second
-	expBackoff.RandomizationFactor = 0.5
-	expBackoff.Multiplier = 2
-	expBackoff.MaxInterval = 15 * time.Second
-	expBackoff.MaxElapsedTime = 2 * time.Minute
-
+	constantBackOff := backoff.NewConstantBackOff(time.Duration(secretsConfig.Timeout) * time.Second)
+	count := 0
 	err = backoff.Retry(func() error {
-		for {
-			log.Info(fmt.Sprintf(messages.CSPFK001I, authn.Config.Username))
-			authnResp, err := authn.Authenticate()
-			if err != nil {
-				return log.RecordedError(messages.CSPFK010E)
-			}
-
-			err = authn.ParseAuthenticationResponse(authnResp)
-			if err != nil {
-				return log.RecordedError(messages.CSPFK011E)
-			}
-
-			err = provideConjurSecrets(accessToken)
-			if err != nil {
-				return log.RecordedError(messages.CSPFK016E)
-			}
-
-			err = accessToken.Delete()
-			if err != nil {
-				return log.RecordedError(messages.CSPFK003E, err.Error())
-			}
-
-			if authnConfig.ContainerMode == "init" || authnConfig.ContainerMode == "application" {
-				log.Info(messages.CSPFK010I)
-				os.Exit(0)
-			}
-
-			// Reset exponential backoff
-			expBackoff.Reset()
-
-			log.Info(messages.CSPFK007I, authn.Config.TokenRefreshTimeout)
-
-			fmt.Println()
-			time.Sleep(authn.Config.TokenRefreshTimeout)
+		err = retryCondition(count, secretsConfig.RetryCountLimit)
+		if err != nil {
+			return err
+		}
+		count++
+
+		log.Info(fmt.Sprintf(messages.CSPFK001I, authn.Config.Username))
```

Comment:
Extract to function from this line to the end of the anonymous function.
Something like: `provideSecretsToTarget` or anything else you like.